ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/RuntimeDataExpireService.java

```diff
@@ -65,27 +65,34 @@ public void doExpire() {
 
   private void expire() {
     List<TableMetadata> tableMetadata = metaService.listTables();
-
+    List<TableIdentifier> optimizeTables = optimizeService.refreshAndListTables();
     // expire and clear table_task_history table
-    tableMetadata.forEach(meta -> {
-      TableIdentifier identifier = meta.getTableIdentifier();
-      try {
-        TableOptimizeRuntime tableOptimizeRuntime =
-            optimizeService.getTableOptimizeItem(identifier).getTableOptimizeRuntime();
-        tableTaskHistoryService.expireTaskHistory(identifier,
-            tableOptimizeRuntime.getLatestTaskPlanGroup(),
-            System.currentTimeMillis() - this.taskHistoryDataExpireInterval);
-      } catch (Exception e) {
-        LOG.error("{} failed to expire and clear table_task_history table", identifier, e);
-      }
-    });
+    if (optimizeTables != null && optimizeTables.size() > 0) {
```

Comment:
Maybe we should just remove `optimizeService.getTableOptimizeItem(identifier).getTableOptimizeRuntime()` from `RuntimeDataExpireService`, because not cleaning the task history with the `latestTaskPlanGroup` 7 days ago brings almost no benefits.